newsroom/auth.py

```diff
@@ -8,15 +14,95 @@
 def login():
     kwargs = {}
     if flask.request.method == 'POST':
-        if flask.request.form.get('username') == 'admin' and flask.request.form.get('password') == 'admin':
-            flask.session['username'] = flask.request.form.get('username')
+        try:
+            credentials = flask.request.form
+
+            user = get_resource_service('auth_user').find_one(req=None, email=credentials.get('email'))
+
+            if not user:
+                raise Exception('Wrong email or password')
+
+            if not user.get('is_enabled'):
+                raise Exception('Account is disabled')
+
+            if not user.get('is_approved'):
+                account_created = user.get('_created')
+
+                if account_created < utcnow() + timedelta(days=-app.config.get('NEW_ACCOUNT_ACTIVE_DAYS', 14)):
+                    raise Exception('Account has not been approved')
+
+            password = credentials.get('password').encode('UTF-8')
+            hashed = user.get('password').encode('UTF-8')
+
+            if not (password and hashed):
+                raise Exception('Wrong email or password')
+
+            if not bcrypt.checkpw(password, hashed):
+                raise Exception('Wrong email or password')
+
+            user = get_resource_service('users').find_one(req=None, _id=user['_id'])
+
+            flask.session['name'] = user.get('name')
             return flask.redirect(flask.url_for('news.index'))
-        kwargs['credentials_error'] = True
-        kwargs['username'] = flask.request.form.get('username', '')
-    return flask.render_template('auth_login.html', **kwargs)
+        except Exception as ex:
+            flask.flash('Invalid login: {} Please try again.'.format(str(ex)))
+            return flask.redirect(flask.url_for('auth.login'))
```

Comment:
I'm not sure if we should redirect on error in general, imo it's better to render the page again and keep the data where appropriate (like email here).